javascript/javascript3/week3/promise to get github repos/main.js

```diff
@@ -0,0 +1,63 @@
+const body = document.querySelector("body");
+const getGitRepos = async function (x) {
+  try {
+    const responseRepos = await fetch(
+      "https://api.github.com/search/repositories?q=user:" + x
+    );
+    if (responseRepos.ok) {
+      const usersData = await responseRepos.json();
+      return usersData;
+    }
+  } catch (error) {
+    console.log(error);
+  }
+};
+getGitRepos("vitalreddym");
+
+(async function () {
+  try {
+    const values = await Promise.all([
+      getGitRepos("vitalreddym"),
+      getGitRepos("alina-kho"),
+      getGitRepos("hassou2n"),
+    ]);
+    console.log(values);
+    for (let i = 0; i < values.length; i++) {
+      const username = document.createElement("ul");
+      username.textContent = values[i].items[0].owner.login + " " + "Repos";
+      for (let j = 0; j < 2; j++) {
+        const repo = document.createElement("li");
+        repo.textContent =
+          values[i].items[j].name + " " + values[i].items[j].git_url;
+        username.appendChild(repo);
+      }
+      body.appendChild(username);
+    }
+  } catch (error) {
+    error;
+  }
+})();
+
+//I tried to do it using just promises without async functions, but I get errors
+
+// const getGithubRepos = function (x) {
+//   return new Promise((resolve) => {
+//     resolve(fetch("https://api.github.com/search/repositories?q=user:" + x));
+//   });
+// };
+
+// getGithubRepos("benna100")
+//   .then((output) => {
+//     return output.json();
+//   })
+//   .catch(console.log.bind(console));
+
+// function get3FriendsGithubRepo() {
+//   const values = Promise.all([
+//     getGithubRepos("benna100"),
+//     getGithubRepos("benna100"),
+//     getGithubRepos("benna100"),
+//   ]).then(() => console.log(values));
+// }
```

Comment:
Pretty close! I think you missed that there will be 3 responses since you did 3 fetch. Here's a way of doing it:
```
function get3FriendsGithubRepo() {
  const values = Promise.all([
    fetch("https://api.github.com/search/repositories?q=user:benna100"),
    fetch("https://api.github.com/search/repositories?q=user:benna100"),
    fetch("https://api.github.com/search/repositories?q=user:benna100"),
  ]).then((responses) => {
    return Promise.all([
      responses[0].json(),
      responses[1].json(),
      responses[2].json(),
    ])
  })
    .then((data) => {
      console.log(data);
      render(data);
    });
}
```
And then you just need a render function and nothing else